components/header/Header.tsx

```diff
@@ -0,0 +1,109 @@
+"use client";
+import React, { useState, useEffect } from "react";
+import Link from "next/link";
+import NavbarCartIcon from "./headerCompontents/NavbarCartIcon";
+import { closeModal } from "@/lib/features/modal/ModalSlice";
+import { useAppDispatch } from "@/lib/hookts";
+import { HiMenu } from "react-icons/hi";
+import CategoriesModal from "@/components/categoriesModal/CategoriesModal";
+import ClosedModal from "../categoriesModal/StripeClosedModal";
+import { AnimatePresence, motion } from "framer-motion";
+import { categoriesSlide } from "@/app/styles/animations/animateVariants";
+
+export default function Header() {
+  const [openCategories, setOpenCategories] = useState<boolean>(false);
+  const [isScrolled, setIsScrolled] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
+
+  const handleOpenModal = () => {
```

Comment:
@Kamildeeal create `useHeader` hook, because this component is getting too complex with all of this logic